### ISSUE TITLE
Plugin asks for Steam Guard Code on every reconnect in combination with Spectrum 2

### ISSUE THREAD

**Issue by Programie** [EionRobb/pidgin-opensteamworks#171]
I'm using this plugin in Spectrum 2 to connect to Steam Chat via XMPP.

Every time I reconnect to the XMPP server, my Steam friends are unreachable and I get a message from the steam mobile contact asking me for the Steam Guard Code. Once I enter the right code, everything works find again, till I disconnect with all clients and reconnect.

I used this plugin with Pidgin before but didn't had this issues. Even after a Pidgin restart, my Steam session stayed valid and I wasn't asked for the Steam Guard Code. Is this some issue related to Spectrum 2 or is it really caused by this plugin?

**Comment by EionRobb** [comment]
We save the steam guard code in the `~/.purple/accounts.xml` file through `purple_account_set_string()` - if spectrum2 doesn't save this file, or overwrites it with its own config, then the guard code will get lost.

I'd like to blame it on spectrum2 since I don't know how that saves config, but maybe if you can get a libpurple debug log to see if it does something with the account settings on disconnect, that might be a bit more helpful than pointing fingers :)

**Comment by Programie** [comment]
There is an accounts.xml file containing an access_token setting.

There is another issue in combination with Spectrum 2 (#69), could that be related to this one?

**Comment by EionRobb** [comment]
Looks like spectrum2 has to add specific support for the guard token:
https://github.com/SpectrumIM/spectrum2/commit/6872e99abbd99a7187084e24cae801a8c6bc3ba0#diff-526d95af29e7782be6a5c02e7f0623ffR1621

I'm wondering if it could also be a captcha request that is unhandled by spectrum2?  Can you attach a libpurple/spectrum debug log?

**Comment by Programie** [comment]
The support for the Steam Guard token should already be part of the Spectrum 2 installation I'm using. Github says, the commit is contained in the 2.0.7 tag and I'm actually running Spectrum 2.0.7.

I don't know how to get a debug log from libpurple in Spectrum. But, the used logger should already log debug messages to the root logger which I configured to write to a log file, but there are only messages for the INFO log level.

But I found something interesting. libpurple says, that the accounts.xml does not exist, which is really true. The file seems to be removed by Spectrum or libpurple and is then recreated. Very strange...
```
31842: 2018-05-18 18:49:12,116 INFO  libpurple: util:File /var/lib/spectrum2/my.steam.service.xmpp.component/accounts.xml does not exist (this is not necessarily an error)
```

So I think this is not an issue with your awesome plugin.
